src/main/java/org/springframework/data/elasticsearch/core/mapping/SimpleElasticsearchPersistentEntity.java

```diff
@@ -138,6 +142,37 @@ public IndexCoordinates getIndexCoordinates() {
 		return resolve(IndexCoordinates.of(getIndexName()));
 	}
 
+	@Override
+	public Set<Alias> getAliases() {
```

Comment:
please add an `aliases` property to the class and either intialize it from the constructor or make it ` Lazy`, check for example the `settingsParameter` property. This should only be evaluated once.